terraform/modules/tf_threat_intel_downloader/dynamodb.tf

```diff
@@ -18,3 +18,80 @@ resource "aws_dynamodb_table" "threat_intel_ioc" {
     Name = "StreamAlert"
   }
 }
+
+// IAM Role: Application autoscalling role
+resource "aws_iam_role" "appautoscaling" {
+  name               = "${var.prefix}_streamalert_appautoscaling"
+  assume_role_policy = "${data.aws_iam_policy_document.appautoscaling_assume_role_policy.json}"
+}
+
+// IAM Policy Doc: Generic Application Autoscaling AssumeRole
+data "aws_iam_policy_document" "appautoscaling_assume_role_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["application-autoscaling.amazonaws.com"]
+    }
+  }
+}
+
+// IAM Role Policy: Allow appautoscaling IAM role to autoscaling DynamoDB table
+resource "aws_iam_role_policy" "appautoscaling" {
+  name   = "DynamoDBAppAutoscalePolicy"
+  role   = "${aws_iam_role.appautoscaling.id}"
+  policy = "${data.aws_iam_policy_document.appautoscaling.json}"
+}
+
+// IAM Policy Doc: Allow autoscaling IAM role to send alarm to CloudWatch
+// and change table settings for autoscaling.
+data "aws_iam_policy_document" "appautoscaling" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "dynamodb:DescribeTable",
+      "dynamodb:UpdateTable",
+      "cloudwatch:PutMetricAlarm",
+      "cloudwatch:DescribeAlarms",
+      "cloudwatch:GetMetricStatistics",
+      "cloudwatch:SetAlarmState",
+      "cloudwatch:DeleteAlarms",
+    ]
+
+    resources = [
+      "${aws_dynamodb_table.threat_intel_ioc.arn}",
+      "${aws_cloudwatch_log_group.threat_intel_downloader.arn}",
+    ]
+  }
+}
+
+resource "aws_appautoscaling_target" "dynamodb_table_read_target" {
+  count              = "${var.autoscale ? 1 : 0}"
+  max_capacity       = 100
+  min_capacity       = 5
+  resource_id        = "${var.prefix}_streamalert_threat_intel_downloader"
```

Comment:
I'd recommend interpolating this from the [dyanmodb_table](https://www.terraform.io/docs/providers/aws/r/dynamodb_table.html) resource instead.  This way, if the name ever changes you don't need to update it in two places